lib/_http_common.js

```diff
@@ -77,6 +77,20 @@ function parserOnHeadersComplete(versionMajor, versionMinor, headers, method,
     parser.incoming.statusMessage = statusMessage;
   }
 
+  // The client made non-upgrade request, and server is just advertising
+  // supported protocols.
+  //
+  // See RFC7230 Section 6.7
+  //
+  // NOTE: RegExp below matches `upgrade` in `Connection: abc, upgrade, def`
+  // header.
+  if (upgrade &&
```

Comment:
@indutny what do you think about only emitting `'upgrade'` if both headers are present in the response? That way the decision could be based purely on the response headers and could be done in `http-parser`. All three use cases of the `upgrade` header require both headers to be present.